File: Week Assignment 6 using Python/8_first_nonrepeat_stream.py
from collections import deque

def FirstNonRepeating(stream):
    q = deque()
    freq = [0]*26
    res = ''
    for ch in stream:
        idx = ord(ch)-97
        freq[idx]+=1
        q.append(ch)
        while q and freq[ord(q[0])-97]>1:
            q.popleft()
        res += q[0] if q else '#'
    return res

File: Week Assignment 6 using C/8_first_nonrepeat_stream.c
#include <stdio.h>
#define MAX 256

void firstNonRepeating(char stream[], int n) {
    int freq[MAX] = {0};
    for (int i = 0; i < n; i++) {
        freq[(int)stream[i]]++;
        char res = '#';
        for (int j = 0; j <= i; j++) {
            if (freq[(int)stream[j]] == 1) { res = stream[j]; break; }
        }
    }
}
